language-support/hs/bindings/test/DA/Ledger/Sandbox.hs

```diff
@@ -7,42 +7,37 @@ module DA.Ledger.Sandbox ( -- Run a sandbox for testing on a dynamically selecte
     Sandbox(..),
     startSandbox,
     shutdownSandbox,
-    withSandbox,
+    withSandbox
     ) where
 
-import Control.Exception (bracket, evaluate, onException)
-import Control.Monad(when)
+import Control.Concurrent (threadDelay)
+import Control.Exception (bracket)
 import DA.Bazel.Runfiles(exe,locateRunfiles,mainWorkspace)
-import DA.Ledger (Port (..), unPort)
-import Data.List (isInfixOf)
-import Data.List.Extra(splitOn)
-import GHC.IO.Handle (Handle, hGetLine)
+import DA.Ledger (Port (..))
+import GHC.IO.Handle (Handle)
+import Safe (readMay)
+import System.IO.Extra (withTempFile)
 import System.FilePath((</>))
-import System.Process (CreateProcess (..), ProcessHandle, StdStream (CreatePipe), createProcess, getPid, interruptProcessGroupOf, proc, waitForProcess)
-import System.Time.Extra (Seconds, timeout)
+import System.Process (CreateProcess (..), ProcessHandle, createProcess, getPid, interruptProcessGroupOf, proc, waitForProcess)
 
 import DA.Ledger.Trace
 
 data SandboxSpec = SandboxSpec {dar :: String}
 
 data Sandbox = Sandbox { port :: Port, proh :: ProcessHandle }
 
-selectedPort :: Int
-selectedPort = 0 --dynamic port selection
 
-sandboxProcess :: SandboxSpec -> IO CreateProcess
-sandboxProcess SandboxSpec{dar} = do
+sandboxProcess :: SandboxSpec -> FilePath -> IO CreateProcess
+sandboxProcess SandboxSpec{dar} portFile = do
     binary <- locateRunfiles (mainWorkspace </> exe "ledger/sandbox/sandbox-binary")
-    pure $ proc binary [ dar, "--port", show selectedPort]
+    pure $ proc binary [ dar, "--port-file", portFile]
 
-startSandboxProcess :: SandboxSpec -> IO (ProcessHandle,Maybe Handle)
-startSandboxProcess spec = do
-    processRecord <- sandboxProcess spec
+startSandboxProcess :: SandboxSpec -> FilePath -> IO (ProcessHandle,Maybe Handle)
+startSandboxProcess spec portFile = do
+    processRecord <- sandboxProcess spec portFile
     (_,hOutOpt,_,proh) <-
         createProcess processRecord {
-        std_out = CreatePipe,
-        std_err = CreatePipe, -- Question: ought the pipe to be drained?
-        create_group = True  -- To avoid sending INT to ourself
+            create_group = True  -- To avoid sending INT to ourself
```

Comment:
I’m not sure I understand this. What exactly is the issue that you are trying to prevent here?